python_modules/dagster/dagster/_core/definitions/asset_key.py

```diff
@@ -197,4 +197,4 @@ def from_user_string(user_string: str) -> "AssetCheckKey":
         return AssetCheckKey(AssetKey.from_user_string(asset_key_str), name)
 
 
-AssetKeyOrCheckKey = Union[AssetKey, AssetCheckKey]
+EntityKey = Union[AssetKey, AssetCheckKey]
```

Comment:
@schrockn `EntityKey` is just `AssetKeyOrCheckKey` in disguise -- I did experiment with making it a true base class, but found that this had worse type-system implications, as it made it impossible to exhaustively check for all valid classes, i.e. you couldn't have functions that looked like:

```python
def to_thing_with_entity(key: EntityKey):
    if isinstance(key, AssetKey):
        ...
    elif isinstance(key, AssetCheckKey):
        ...
```

because the type system wouldn't know that those were the only two valid types of entity keys. So you'd need to toss an "`else: raise`" at the bottom, which would just be a landmine if we ever add anything to this union.

this way, if we add any new entity types, pyright can just tell us all the places that we need to update.

Reply:
Got it. I'm happy you got to this conclusion as I was quite skeptical of the subclasses approach anyways.